docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,111 +1,322 @@
+
+```kotlin
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
+
+kotlin {
+    // Enable the default target hierarchy:
+    targetHierarchi.default()
+
+    jvm()
+    iosArm64()
+    // the rest of the necessary targets...
+
+    sourceSets {
+        val commonMain by getting
+
+        val jvmAndNativeMain by creating {
+            dependsOn(commonMain)
+        }
+
+        val nativeMain by getting {
+            dependsOn(jvmAndNativeMain)
+        }
+
+        val jvmMain by getting {
+            dependsOn(jvmAndNativeMain)
+        }
+    }
+}
 ```
-{initial-collapse-state="collapsed"}
 
-Instead of doing this, you can create a hierarchical structure
-with [target shortcuts](multiplatform-share-on-platforms.md#use-target-shortcuts)
-available for typical multi-target scenarios, or you can manually declare and connect the source sets. For example, you
-can create two iOS targets and a shared source set with the `ios()` shortcut:
+> We're currently working on API for creating your own target hierarchies. It should be useful for projects
+> which hierarchy configurations differ a lot from the default template.
+>
+> At the moment, this API is not ready yet. But if you're eager to try it,
+> look into the `targetHierarchy.custom { ... }` block and the declaration of `targetHierarchy.default()` as an example.
+> Keep in mind that this API is still in development. It might be under-tested, and can change dramatically in further releases.
+> 
+{type="tip"}
+
+## Target shortcuts
+
+The Kotlin Multiplatform plugin provides some pre-defined target shortcuts for creating structures of common target
+combinations:
+
+| Target shortcut | Targets                                      |
+|-----------------|----------------------------------------------|
+| `ios`           | `iosArm64`, `iosX64`                         |
+| `watchos`       | `watchosArm32`, `watchosArm64`, `watchosX64` |
+| `tvos`          | `tvosArm64`, `tvosX64`                       |
+
+All shortcuts create similar hierarchical structures in the code. For example, you can create a multiplatform project
+with two iOS-related targets, `iosArm64` and `iosX64` and a shared source set with the `ios()` shortcut:
 
 ```kotlin
 kotlin {
     ios() // iOS device and simulator targets; iosMain and iosTest source sets
 }
 ```
 
-The Kotlin toolchain will provide the correct default dependencies and locate the API surface area available in the shared
-code. This prevents cases like the use of a macOS-specific function in code shared for Windows.
+In this case, the hierarchical structure includes intermediate source sets `iosMain` and `iosTest`,
+which are used by the platform-specific source sets:
+
+![Code shared for iOS targets](iosmain-hierarchy.png)
+
+The resulting hierarchical structure will look like this:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+kotlin {
+    sourceSets {
+        val commonMain by sourceSets.getting
```

Comment:
This code is already inside `sourceSets{}`. So, in `sourceSets.getting` call `sourceSets.` part might be omitted.